mne/io/eeglab/eeglab.py

```diff
@@ -226,6 +233,12 @@ class RawEEGLAB(_BaseRaw):
         Names or indices of channels that should be designated
         EOG channels. If 'auto', the channel names containing
         ``EOG`` or ``EYE`` are used. Defaults to empty tuple.
+    event_id : dict
+        By default, events are read from the input file by dropping every
+        non-integer part of events containing integers, and completely
+        dropping any events without integer parts. If non-integer events
+        should be read, this should be a dict mapping from their names to
+        integers, e.g. dict(fmri_scan_onset=199, recording_start=255).
```

Comment:
I see 3 options (after reading maybe too quickly):

1/ we allow an auto mode for event_id in read_raw_eeglab that just sorts the keys and assigns unique numbers to all of them.
2/ we do it in 2 steps. A function to generate an event_id from a set file (as described above) that is then passed to the reader. Benefit is that  it is more explicit.
3/ we let the user manually define his event_id.

wdyt?